src/Symfony/Component/DependencyInjection/Attribute/When.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+/**
+ * An attribute to tell under which environement this class should be autoconfigured.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
+class When
+{
+    public function __construct(
+        public string $env,
```

Comment:
```suggestion
        public string $environment,
```

For consistency. We have:
* `Kernel::__construct(string $environment, bool $debug)`
* `Kernel::getEnvironment()`
* `%kernel.environment%`